src/PkgAuthentication.jl

```diff
@@ -356,19 +394,32 @@ function step(state::ClaimToken)::Union{ClaimToken, HasNewToken, Failure}
     sleep(state.poll_interval)
 
     output = IOBuffer()
-    data = JSON.json(Dict(
-        "challenge" => state.challenge,
-        "response" => state.response,
-    ))
-    response = Downloads.request(
-        string(state.server, "/claimtoken"),
-        method = "POST",
-        input = IOBuffer(data),
-        output = output,
-        throw = false,
-    )
+    is_device = state.response isa Dict{String, Any} && get(state.response, "client_id", nothing) == "device"
+    if is_device
+        output = IOBuffer()
+        response = Downloads.request(
+            joinpath(state.server, "dex/token"),
```

Comment:
Discover token endpoint from well-known config instead of hardcoding.

Also, have two internal functions for clarity? One for device flow, and another for challenge-response flow.

Reply:
Done in https://github.com/JuliaComputing/PkgAuthentication.jl/pull/42/commits/79e417d17ba77db3945a743853aa08d7a126385e